wilsonChan/assignments/slidingwindow/lc53/lc53.py

```diff
@@ -1,13 +1,42 @@
-def max_sub_array_of_size_k(k, arr):
+# Problem Statement #
 
+# Given an array of positive numbers and a positive number ‘k,’ find the maximum sum of any contiguous subarray of size ‘k’.
 
-def main():
-  print("Maximum sum of a subarray of size K: " + str(max_sub_array_of_size_k(3, [2, 1, 5, 1, 3, 2])))
-  print("Maximum sum of a subarray of size K: " + str(max_sub_array_of_size_k(2, [2, 3, 4, 1, 5])))
+'''
+  #This for loop on lines 25-28 is giving me some trouble. The output only returns the
+  sum of the first window:
+      Maximum sum of a subarray of size K: 8
+      Maximum sum of a subarray of size K: 5
 
+  I'm trying to remove the previous element on the left side of the current window 
+  and add the next element of on the right side of the current window
+'''
 
-main()
+def max_subarray_size_k(k, arr):
+
+  #Initialize variables
+  maxSum = 0
+  current = 0
+
+  #The sum of the first window based on provided k value
+  for i in range(k):
+    maxSum += arr[i]
+
+  for i in range(k):
+    current += arr[i] - arr[i-k]
+    maxSum = max(maxSum, current)
+  return maxSum
 
+  #Dealing with empty array
+  if(n == 0 or k > n or k < 1):
+    return 0
```

Comment:
In this case, you want to loop through the array once, and while looping check to see if we have reached the number k (because that means we have added `k` elements to the current sum. Then compare the current sum to the max sum, and if bigger, replace the max sum. Then slide the window (subtract the value at the start of the subarray, increment the start pointer).

Try this problem again this weekend, and make sure if you get a working solution to write it twice to build that muscle memory!